[chen-jerry-junior] iP

chen-jerry-junior opened the pull request:
# Duke :laughing:
<blockquote>
善问者，如攻坚木，先其易者，后其节目 ——《礼记·学记》
</blockquote>
A Personal Assistant Chatbot that can help YOU to keep track of anythings. 

---

- text-based
- easy to learn
- super fast to use

##Features which Duke holds:
1. __FREE__
2. Managing tasks
3. Managing deadlines
4. Reminders (coming soon)

---

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public static void main(String[] args) {
    new Duke().run();
}
```

Commit:
Add Gradle support

chen-jerry-junior committed:
level 1

chen-jerry-junior committed:
level 2

chen-jerry-junior committed:
level 3

chen-jerry-junior committed:
level 4

chen-jerry-junior committed:
A-TextUiTesting

chen-jerry-junior committed:
level 5

chen-jerry-junior committed:
level 6

chen-jerry-junior committed:
level 6 improved

chen-jerry-junior committed:
level 7

chen-jerry-junior committed:
level 8

chen-jerry-junior committed:
Merge branch 'branch-Level-8'

chen-jerry-junior committed:
add more OOP

chen-jerry-junior committed:
add packages

chen-jerry-junior committed:
Merge branch 'add-gradle-support'

chen-jerry-junior committed:
add JUnit

chen-jerry-junior committed:
modify build.gradle

chen-jerry-junior committed:
add coding standard and checkstyle

zrei commented on src/main/java/duke/Deadlines.java:
```diff
@@ -0,0 +1,54 @@
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class Deadlines extends Task {
```
Shouldn't the class name be `Deadline` instead?

zrei commented on src/main/java/duke/Deadlines.java:
```diff
@@ -0,0 +1,54 @@
+        return this.date;
+    }
+
+    public String toString() {
```
Shouldn't there be an `@Override` annotation here?

zrei commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,40 @@
+package duke;
+
+import duke.commands.*;
```
Shouldn't the import statements be explicit?

zrei commented on src/main/java/duke/Deadlines.java:
```diff
@@ -0,0 +1,54 @@
+        DateTimeFormatter df = DateTimeFormatter.ofPattern("MMM dd yyyy");
+        String result = "";
+        if (this.getStatus()) {
+            result += "[D][X] " + this.getName() + " (by: ";
```
Maybe you can extricate the setting of `[ ]` and `[X]` to another variable?

Such as
```java
String taskStatus = this.getStatus() ? "[X] " : "[ ] ";
result += "[D] " + taskStatus + this.getName() + " (by: ");
```

zrei commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,66 @@
+        String arr1[] = input.split("/");
+        String arr2[] = arr1[0].split(" ");
+        switch (arr2[0]) {
+            case "list":
```
Shouldn't the case clauses have no indentation?

Such as
```java
switch (arr2[0]) {
case "list":
    return new ListCommand();
```

zrei commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,66 @@
+                if (arr2.length < 2) {
+                    throw new DukeException("OOPS!!! The description of a unmark cannot be empty.");
+                }
+                int No = Integer.parseInt(arr2[1]);
```
Shouldn't this variable be named in camelCase? i.e. `no`

Maybe you could consider renaming this variable to something like `index` to make its purpose clearer.

zrei commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,30 @@
+
+public class Task implements Serializable {
+    private String name;
+    private boolean status;
```
Shouldn't this variable be named to sound like a boolean?

Maybe you could rename it to something like `isDone`

zrei commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,30 @@
+        this.status = status;
+    }
+
+    public boolean getStatus() {
```
Similar to above, maybe you could rename this method to something like `isDone`

zrei commented on src/main/java/duke/commands/DeleteCommand.java:
```diff
@@ -0,0 +1,34 @@
+
+public class DeleteCommand extends Command {
+
+    private int deleteNumber;
```
Maybe you can consider renaming this variable to `indexToBeDeleted` to make its purpose clearer.

zrei commented on src/main/java/duke/commands/Command.java:
```diff
@@ -0,0 +1,19 @@
+
+public class Command {
+
+    public void execute(TaskList list, Storage store, Ui ui) throws DukeException {
```
Maybe you can consider making this method abstract and making the `Command` class abstract.

zrei commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,40 @@
+        tasks = storage.load();
+    }
+
+    public void run() {
```
Shouldn't there be a header comment here?

zrei reviewed (commented):
Good job on the neat code! :+1:

chen-jerry-junior committed:
add JavaDoc

chen-jerry-junior committed:
level 9

chen-jerry-junior committed:
modified JavaDoc

chen-jerry-junior committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/DukeException.java
#	src/main/java/duke/Todos.java

fix1

chen-jerry-junior committed:
modified fix1

chen-jerry-junior committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	data/duke.txt
#	src/main/java/duke/Parser.java

fix2

chen-jerry-junior committed:
modified master branch after merging level 9

chen-jerry-junior committed:
add CheckStyle

Mr-Teal commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,57 @@
+     *
+     * @return the task name.
+     */
+    public String toString() {
```
The function of toString() seems totally the same as getName() above. Maybe you can delete one of them.

Mr-Teal commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,67 @@
+     *
+     * @return the user command
+     */
+    public String getUserCommand() {
```
I personally consider this method not necessary, but separating the situatiton of empty input and wrong input is a good idea.

Mr-Teal commented on src/main/java/duke/commands/FindCommand.java:
```diff
@@ -0,0 +1,60 @@
+     */
+    @Override
+    public void execute(TaskList list, Storage store, Ui ui) throws DukeException {
+        ui.showMessage("Here are the matching tasks in your list:");
```
You may put this line below as it may seem strange when the Duke want to show the matching tasks it finds but then it says no tasks in the list.

Mr-Teal commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,57 @@
+     *
+     * @param status
+     */
+    public void setStatus(boolean status) {
```
This method can be separated into two different methods. One only for Mark and the other only for Unmark. This is to be easy to read for the code readers (also a coding habit from CS2030S).

Mr-Teal reviewed (commented):
No serious problems. Good work!

chen-jerry-junior committed:
add GUI

nramapurath commented on src/main/java/duke/Deadlines.java:
```diff
@@ -0,0 +1,78 @@
+     */
+    public Deadlines(String name, String sTime) throws DukeException {
+        super(name);
+        DateTimeFormatter dtf = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+        DateTimeFormatter df = DateTimeFormatter.ofPattern("yyyy-MM-dd");
```
Perhaps you could use more descriptive variable names here rather than dtf or df?

nramapurath commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,42 @@
+
+
+    /**
+     * You should have your own function to generate a response to user input.
+     * Replace this stub with your completed method.
+     */
```
Maybe you could replace this placeholder comment with information more specific to the function the way you've implemented it?

nramapurath commented on src/main/java/duke/Events.java:
```diff
@@ -0,0 +1,53 @@
+/**
+ * A task type that the chatting bot can create.
+ */
+public class Events extends Task {
```
Shouldn't the class name here be `Event` as the plural form would indicate a collection of Event objects?

nramapurath commented on src/main/java/duke/Todos.java:
```diff
@@ -0,0 +1,30 @@
+/**
+ * A task type that the chatting bot can create.
+ */
+public class Todos extends Task {
```
Maybe you could rename the class to `ToDo` (as plurals would lead people to think of a collection of objects)?

nramapurath commented on src/main/java/duke/commands/DeadlineCommand.java:
```diff
@@ -0,0 +1,70 @@
+    /**
+     * The method that includes the execution of the command.
+     *
+     * @param list
+     * @param store
```
Perhaps these parameters could use a one-liner description as in the example given in the [coding standards](https://se-education.org/guides/conventions/java/intermediate.html#comments)?

nramapurath reviewed (commented):
Overall, I like the code! It is easy to read for the most part, with good use of whitespace for readability and few line breaks needed. There is a clear division of what each class's role is and it shows the time and effort put into the code. Good job!

chen-jerry-junior committed:
Add assertion statement

There are many commands that can cause the duke to misbehave.

Assertion statements can help to regulate the command input.

Let's
* add assertion to regulate mark command
* add assertion to regulate unmark command

chen-jerry-junior committed:
Modify the time format in Events.java

The time format in event command is string, which can not be calculated.

I introduced LocalDateTime to regulate the time input following the event command. I also changed the variable name to improve the code quality.

Let's
* add LocalDate to regulate event command
* add LocalTime to regulate event command

chen-jerry-junior committed:
Replace the traversing operation of arrayList with stream.filter() method.

The loop to iterate the encapsulated arrayList is to verbose.

I imported streams to help me filter the arrayList to find the target task.

Let's
* add Streams in MarkCommand
* add Streams in UnmarkCommand
* add Streams in FindCommand

chen-jerry-junior committed:
Merge pull request #2 from chen-jerry-junior/branch-A-Assertions

Add assertion statement

chen-jerry-junior committed:
delete the storage

chen-jerry-junior committed:
Merge branch 'branch-A-CodeQuality'

# Conflicts:
#	data/duke.txt

chen-jerry-junior committed:
ignored the storage data/duke.txt

chen-jerry-junior committed:
Merge branch 'branch-A-Streams'

# Conflicts:
#	data/duke.txt

chen-jerry-junior committed:
Add CI

chen-jerry-junior committed:
Revert "Add CI"

This reverts commit 47b82167224dd5cf7bcc09f5d899c43156e0f6f6.

chen-jerry-junior committed:
Add extention C-DetectDuplicate

chen-jerry-junior committed:
revise according to iP code quality feedback

chen-jerry-junior committed:
Allow resizing of the Window, and ensure the content resize appropriately as the Window changes size

chen-jerry-junior committed:
Add a personality to duke

chen-jerry-junior committed:
Add a better GUI

chen-jerry-junior committed:
Add a greeting according to the previous personality

chen-jerry-junior committed:
Merge branch 'branch-A-Personality'

chen-jerry-junior committed:
Modify MainWindow.java after merging A-personality and A-BetterGui

chen-jerry-junior committed:
Add a ExceptionCommand class to handle more Exceptions

chen-jerry-junior committed:
Modify more error handling

chen-jerry-junior committed:
Add userguide

chen-jerry-junior committed:
Modify the .gradle to rename the .jar

chen-jerry-junior committed:
Upload the icon

chen-jerry-junior committed:
Upload Ui.png

chen-jerry-junior committed:
A Jnit test

chen-jerry-junior committed:
Fix the javadoc and Find command

chen-jerry-junior committed:
Improve the UI

chen-jerry-junior committed:
Update README

Expanded README to include project details and usage instructions for Space Lin.
